### PR TITLE
Component communication

### PR DESCRIPTION
**Pull request by Supereg**
# Component communication

## :recycle: Current situation & Problem
Currently, there is no easy way for a `Component` to access features or information provided by another `Component`.

I illustrate the usefulness using a real-world example we are currently facing with `SpeziAccount`. A `AccountConfiguration` is used by the user to configure the `SpeziAccount` subsystem which in turn does everything necessary to set up the request `AccountServices` to be accessible by the App's views. While one may just provide off-the-shelve `AccountServices` using the initializer of the configuration object, there might be instances where an third-party `AccountService` is controlled and configured by a `Component`. In such cases it should be possible for the user to place the `Component` in the `Configuration` closure as usual with the `AccountConfiguration` querying `AccountService` instances from all configured `Components`.

```swift 
let config = Configuration(standard: TestAppStandard()>) {
    AccountConfiguration {
        SimpleAccountService()
    }

    FirebaseComponent() // a component preforming necessary initialization steps for its AccountService
}
```

## :bulb: Proposed solution
In this PR we introduce the `@Provide` and `@Collect` property wrappers that can be used to pass around information or functionality between `Components`. The `@Provide` property wrapper is used to supply a single `Value` (or `Value?` if it e.g. only supplies it conditionally) or  multiple values (`[Value]`) to be collected by other `Components`.
Another `Component` can then use `@Collect` with a type of `[Value]` to collect all the provided items.

Here is a short code example:
```swift
class FirebaseComponent<ComponentStandard: Standard>: Component {
    @Provide private var accountService: any AccountService

    init() {
        // @Provide have to be initialized before the components configure() method is called
        self.accountService = FirebaseAccountService() // may also be initialize inline
    }

    func configure() {
        // do firebase initializations 
        accountService.configure() // as its a reference type, we can adjust state
    }
}

class AccountConfiguration<ComponentStandard: Standard>: Component {
    @Collect private var accountServices: [any AccountService]

    func configure() {
        // @Collect can be used within the configure method
        // ...
    }
}
```

While the example shows the usage of a reference type, this feature can also easily be used with value types (though then, no modifications post the collection phase are possible).

## :gear: Release Notes
* Added `@Provide` and `@Collect` property wrappers for easy to use inter-`Component` communication
* Added a generalized and reusable `SharedRepository` pattern to easily implement typed collections.

### Shared Repository Pattern

This PR additionally introduces the concept of the `SharedRepository` software pattern based on Buschmann et al. (_Pattern-Oriented Software Architecture: A Pattern Language for Distributed Computing_). We provide a variety of different types of `KnowledgeSource`s which are stored in `SharedRepository` implementations. Each shared repository defines a `SharedRepositoryAnchor` it expects from an implementing `KnowledgeSource`.

This instantiation of the shared repository pattern replaces the previous `TypedCollection` implementation. Further, it will be of equal use for `SpeziAccount` (storing arbitrary account details). The current draft PR introduces a similar concept. This PR helps to generalize this software pattern and increase code reuse across the framework ecosystem.

### Alternative Considered

As with the previous `TypeCollection` implementation, a `SharedRepository` allows for a `collect(allOf:)` query which is, e.g., internally used to query all `Components` that conform to `LifecycleHandler`. Instead of the proposed system, we could have created a property wrapper that just collects all components that conform to a given protocol. I decided against this approach considering the following points:
* The alternative solution would not allow to prohibit propagation of information (only by the visibility of a protocol itself).
* The alternative solution would allow to circumvent the current `@Dependency` system (or we would need to consider '@Collect` declarations in the execution order).
* The alternative system would be limited to just retrieve protocol conformances.
* The proposed solution decouples propagated information from the `Component` by default allowing for arbitrary information (value types, reference types, information, functionality).

### Related PRs
* https://github.com/StanfordSpezi/SpeziAccount/pull/7

### Testing
Test cases were adapted but still need to be extended for newly added code.

### Reviewer Nudging
There are two separate parts to the PR:
* The `SharedRepository` data structure
* The `Collect` and `Provide` property wrappers. You may look at the tests to easily glance what their use case is.

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: Sources/Spezi/Module/Capabilities/Communication/CollectPropertyWrapper.swift [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        }
+
+        return values
+    }
+
+    /// Initialize a new `@Collect` property wrapper.
+    public init() {}
+}
+
+/// A adopter of this protocol is a property of a ``Component`` that provides mechanisms to retrieve
```
Comment (PSchmiedmayer) [inline Sources/Spezi/Module/Capabilities/Communication/CollectPropertyWrapper.swift:38]: Not sure if this is in the right place here?

File: Sources/Spezi/Module/Capabilities/Communication/CollectPropertyWrapper.swift [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    /// retrieving them from the central ``SpeziStorage`` repository.
+    ///
+    /// **Retrieving Data**
+    /// ``Component/Collect`` retrieves data provided through the ``Component/Provide`` property wrapper.
+    /// You declare `@Collect` as an Array with a given type. The type is used to match `@Provide` properties.
+    ///
+    /// Below is an example where the `ExampleComponent` collects an array of `Numeric` types from all other `Components`.
+    ///
+    /// ```swift
+    /// class ExampleComponent<ComponentStandard: Standard>: Component {
```
Comment (PSchmiedmayer) [inline Sources/Spezi/Module/Capabilities/Communication/CollectPropertyWrapper.swift:55]: Thank you for the example!

File: Sources/Spezi/Module/Capabilities/Communication/CollectPropertyWrapper.swift
Comment (PSchmiedmayer) [inline Sources/Spezi/Module/Capabilities/Communication/CollectPropertyWrapper.swift:1]: Used this file as an example about the new line usage in the project, the comments can be applied to the complete PR.

File: Sources/Spezi/Module/Capabilities/Communication/ProvidePropertyWrapper.swift [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+    /// Initialize a new `@Provide` property wrapper.
+    /// - Parameter value: The initial value.
+    public init(wrappedValue value: Value) {
+        self.storedValue = value
+    }
+}
+
+
+extension Component {
+    /// The `@Provide` property wrapper can be used to communicate data with other ``Component``s by inserting
```
Comment (PSchmiedmayer) [inline Sources/Spezi/Module/Capabilities/Communication/ProvidePropertyWrapper.swift:55]: Amazing example that proves a lot of context, very nice; thank you for that!

File: Sources/Spezi/Configuration/Component.swift [hunk 4]
```diff
@@ -47,19 +47,28 @@
...
-public protocol Component<ComponentStandard>: AnyObject, TypedCollectionKey {
+public protocol Component<ComponentStandard>: AnyObject, KnowledgeSource<SpeziAnchor> {
     /// A ``Component/ComponentStandard`` defines what ``Standard`` the component supports.
     associatedtype ComponentStandard: Standard
+
+    /// The `prepare` method is an optional method you may use to prepare your ``Component`` before it is configured (see ``configure()``).
+    ///
+    /// More specifically, the `prepare` method is called once the ``Component/StandardActor`` property is accessible and the ``Component/Provide``
+    /// properties are not yet collected.
+    func prepare()
```
Comment (PSchmiedmayer) [inline Sources/Spezi/Configuration/Component.swift:58]: I was looking at the PR and the usage of the `prepare()` function and was wondering if the main reason behind using it compared to using the initializer is that we already have the standard accessible, allowing some decisions based on the standard provided and some communication with the standard instance?

File: Sources/Spezi/Module/Capabilities/Communication/StorageValueCollector.swift [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+//
+// SPDX-License-Identifier: MIT
+//
+
+
+/// A adopter of this protocol is a property of a ``Component`` that provides mechanisms to retrieve
+/// data provided by other ``Component``s.
+///
+/// Data requested through a Storage Value Collector might be provided through a ``StorageValueProvider``.
+public protocol StorageValueCollector {
```
Comment (PSchmiedmayer) [inline Sources/Spezi/Module/Capabilities/Communication/StorageValueCollector.swift:14]: Do we want to explore this in the documentation? If not, we might want to add an underscore before the Type name. Might also apply to some other types that are currently public. Also ensures that they don't show up in autocomplete outside of the target.

File: Sources/Spezi/Module/Capabilities/Communication/StorageValueProvider.swift [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+//
+// SPDX-License-Identifier: MIT
+//
+
+
+/// A adopter of this protocol is a property of a ``Component`` that provides mechanisms to communicate
+/// data with other ``Component``s.
+///
+/// Data provided through a Storage Value Provider can be retrieved through a ``StorageValueCollector``.
+public protocol StorageValueProvider {
```
Comment (PSchmiedmayer) [inline Sources/Spezi/Module/Capabilities/Communication/StorageValueProvider.swift:14]: Similar question here, if we would also like to underscore this type.

File: Sources/Spezi/Module/Capabilities/Dependencies/DynamicDependenciesPropertyWrapper.swift [hunk 7]
```diff
@@ -27,18 +27,18 @@ public class _DynamicDependenciesPropertyWrapper<S: Standard>: DependencyDescrip
...
     public func gatherDependency(dependencyManager: DependencyManager<S>) {
         for componentProperty in componentProperties {
-            garther(componentProperty, in: dependencyManager)
+            gather(componentProperty, in: dependencyManager)
         }
     }
-    
-    private func garther<D: ComponentDependency>(_ componentProperty: D, in dependencyManager: DependencyManager<S>) where D.PropertyStandard == S {
+
+    private func gather<D: ComponentDependency>(_ componentProperty: D, in dependencyManager: DependencyManager<S>) where D.PropertyStandard == S {
```
Comment (PSchmiedmayer) [inline Sources/Spezi/Module/Capabilities/Dependencies/DynamicDependenciesPropertyWrapper.swift:41]: Thank you!!

File: Sources/Spezi/Utilities/Component+Properties.swift [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+
+extension Component {
+    func retrieveProperties<Value>(ofType type: Value.Type) -> [Value] {
```
Comment (PSchmiedmayer) [inline Sources/Spezi/Utilities/Component+Properties.swift:11]: Nice 🚀

File: Sources/Spezi/Utilities/AnyOptional.swift [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+//
+// SPDX-License-Identifier: MIT
+//
+
+/// A type erased `Optional`.
+///
+/// This is useful to unwrapping, e.g.,  generics or associated types by declaring an extension under the condition of
+/// ``AnyOptional`` conformance. This allows in the implementation of the extension to access the underlying
+/// ``Wrapped`` type of an `Optional`, essentially unwrapping the optional type.
+public protocol AnyOptional {
```
Comment (PSchmiedmayer) [inline Sources/Spezi/Utilities/AnyOptional.swift:15]: Haha, remember this code and some of the decisions that went into writing it, good old times 😅

File: Sources/Spezi/Spezi/Spezi.swift [hunk 10]
```diff
@@ -63,23 +72,51 @@ public actor Spezi<S: Standard>: AnySpezi, ObservableObject {
     init(
         standard: S,
         components: [any Component<S>],
-        _ logger: Logger = Logger(subsystem: "edu.stanford.spezi", category: "spezi")
+        _ logger: Logger = speziLogger
```
Comment (PSchmiedmayer) [inline Sources/Spezi/Spezi/Spezi.swift:75]: We should ideally also explore the logger to some components using a knowledge source and maybe even a nice property wrapper that is encourage to use for logging in Spezi modules (apps should ideally have their own logging instance). But might be the content of a different PR if this goes too far.

File: Sources/Spezi/SharedRepository/Builtin/BuiltinRepository.swift [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+    associatedtype Anchor
+    typealias Container = [ObjectIdentifier: AnyRepositoryValue]
+
+    var storage: Container { get set }
+}
+
+
+// extension provide default implementation with `0` suffix, as these all have internal visibility.
+extension BuiltinRepository {
+    func get0<Source: KnowledgeSource<Anchor>>(_ source: Source.Type) -> Source.Value? {
```
Comment (PSchmiedmayer) [inline Sources/Spezi/SharedRepository/Builtin/BuiltinRepository.swift:21]: Nice, I get some standard lib vibes here 😄

File: Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+
+
+/// A ``ValueRepository`` that allows to store any ``KnowledgeSource``s.
+public typealias UniversalValueRepository = ValueRepository<Any>
+
+
+/// A ``SharedRepository`` implementation that itself is a value type.
+///
+/// - Note: The ``ValueRepository`` is considered `Sendable` if the ``RepositoryAnchor`` conforms to `Sendable`.
+public struct ValueRepository<Anchor>: SharedRepository, BuiltinRepository {
```
Comment (PSchmiedmayer) [inline Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift:17]: I don't see any usage of the `ValueRepository`, I suspect this is more meant for other modules to use? We might want to either add some article in the DocC docs on how to use the shared repository mechanism (IMO preferred) or at least add some comment here demonstrating this usage so developers have some examples how to use these mechanisms.

File: Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+        set0(source, value: newValue)
+    }
+
+    public func collect<Value>(allOf type: Value.Type) -> [Value] {
+        collect0(allOf: type)
+    }
+}
+
+
+extension ValueRepository: @unchecked Sendable where Anchor: Sendable {}
```
Comment (PSchmiedmayer) [inline Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift:36]: Should we somehow add a locking mechanism to enforce some Sendable mechanisms here? 
I might miss something, I would say that by being a struct we should not really have a problem here and it is more up to `Container` to enforce anything here and it is not modifiable anyways. Am I on the correct path? Might be good to add a comment above the `@unchecked` conformance to ensure that we document the reasoning here if we don't use a locking mechanism.

File: Sources/Spezi/SharedRepository/KnowledgeSource.swift [hunk 14]
```diff
@@ -0,0 +1,67 @@
...
+
+/// A `KnowledgeSource` serves as a typed key for a ``SharedRepository`` implementation.
+///
+/// A ``KnowledgeSource`` is anchored to a given ``RepositoryAnchor`` and defines a ``Value`` type which by default
+/// is set to `Self`.
+public protocol KnowledgeSource<Anchor> {
+    /// The type of a value this `KnowledgeSource` represents.
+    associatedtype Value = Self
+    /// The ``RepositoryAnchor`` to which this `KnowledgeSource` is anchored to.
+    associatedtype Anchor: RepositoryAnchor
```
Comment (PSchmiedmayer) [inline Sources/Spezi/SharedRepository/KnowledgeSource.swift:48]: I like the anchor mechanism, also feels familiar and with some of the stuff we did in Apodini.

File: Sources/Spezi/SharedRepository/SharedRepository.swift [hunk 15]
```diff
@@ -0,0 +1,135 @@
...
+// SPDX-License-Identifier: MIT
+//
+
+
+/// This protocol provides a common interface for a storage mechanism that allows multiple entities
+/// to access, provide and modify shared data.
+///
+/// A `SharedRepository` allows to easily integrate application functionality with a data-driven control flow or
+/// applications that operate on the same data, but do not share the same processing workflow or are split accross
+/// multiple software systems (Buschmann et al. _Pattern-Oriented Software Architecture: A Pattern Language for Distributed Computing_).
```
Comment (PSchmiedmayer) [inline Sources/Spezi/SharedRepository/SharedRepository.swift:15]: Bernd would be proud.

File: Sources/Spezi/SharedRepository/SharedRepository.swift [hunk 16]
```diff
@@ -0,0 +1,135 @@
...
+    }
+
+    /// Default subscript implementation delegating to ``get(_:)`` or providing a ``DefaultProvidingKnowledgeSource/defaultValue``.
+    public subscript<Source: DefaultProvidingKnowledgeSource<Anchor>>(_ source: Source.Type) -> Source.Value {
+        self.get(source) ?? source.defaultValue
+    }
+
+    /// Default subscript implementation delegating to ``get(_:)`` or calling ``ComputedKnowledgeSource/compute(from:)``
+    /// and storing the result.
+    public subscript<Source: ComputedKnowledgeSource<Anchor>>(_ source: Source.Type) -> Source.Value {
```
Comment (PSchmiedmayer) [inline Sources/Spezi/SharedRepository/SharedRepository.swift:108]: Nice, like the usage of subscripts here.

**Review by PSchmiedmayer — APPROVED** [review]
Thank you so much for all the changes there @Supereg, a huge step forward 🚀

I had a few comments here and there. Overall the PR looks great and feel free to merge it after addressing some of the comments and iterating on some of the features. Thank for all the work and time that goes into this project. 

Very cool to see some of the @Apodini aspects making a reappearance here; more and more of the learnings from that project are applied to @StanfordSpezi, very nice!

File: Sources/Spezi/Module/Capabilities/Communication/StorageValueCollector.swift (hunk 5 above)
Comment (Supereg) [reply Sources/Spezi/Module/Capabilities/Communication/StorageValueCollector.swift:14]: Probably the use case for extending this from other modules is rather small. We can hide it from documentation for now 👍

File: Sources/Spezi/Utilities/AnyOptional.swift (hunk 9 above)
Comment (Supereg) [reply Sources/Spezi/Utilities/AnyOptional.swift:15]: Yes. I had the gut feeling that I had written that code before haha.
I tried several approaches to avoid having those protocols implemented, but seems like the most sensible way to me.
I thought I just make them public?

File: Sources/Spezi/Spezi/Spezi.swift (hunk 10 above)
Comment (Supereg) [reply Sources/Spezi/Spezi/Spezi.swift:75]: I like the idea of a `@Logger` property wrapper (using a dedicated property wrapper probably allows for better customization of a `subsystem` and a `category`).
I think that suites a separate PR a bit better. But we could store the global `speziLogger` using a module internal KnowledgeSource.

File: Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift (hunk 12 above)
Comment (Supereg) [reply Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift:17]: This is something I will use e.g. in `SpeziAccount`. A `ValueRepository` (having Value semantics) is better suited for controlling write access (through property mutability, similar to a regular Array or Dictionary). As this is probably a typical suite case I put that implementation here.

File: Sources/Spezi/SharedRepository/KnowledgeSource.swift (hunk 14 above)
Comment (Supereg) [reply Sources/Spezi/SharedRepository/KnowledgeSource.swift:48]: I think in Apodini we also had some Anchoring technique but it was quite different 😅  But there is definitely inspiration.

File: Sources/Spezi/Module/Capabilities/Communication/CollectPropertyWrapper.swift (hunk 1 above)
Comment (Supereg) [reply Sources/Spezi/Module/Capabilities/Communication/CollectPropertyWrapper.swift:37]: Upise 🙈

File: Sources/Spezi/Module/Capabilities/Communication/CollectPropertyWrapper.swift [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+
+
+/// Refer to the documentation of ``Component/Collect``.
+@propertyWrapper
+public class _CollectPropertyWrapper<Value> {
+    // swiftlint:disable:previous type_name
+    // We want the type to be hidden from autocompletion and documentation generation
+
+    private var injectedValues: [Value]? // swiftlint:disable:this discouraged_optional_collection
+
```
[the remark this answers is not in the document]
Comment (Supereg) [reply Sources/Spezi/Module/Capabilities/Communication/CollectPropertyWrapper.swift:19]: Do you think that this is feasible to be implemented as a custom `SwiftLint` (cause it's entirely whitespace related). SwiftLint already has rules about ordering  members and also for certain whitespace checks.
Cause currently this is really hard to enforce as it is really easy to overlook, therefore quite cumbersome.

[the remark this answers is not in the document]
Comment (PSchmiedmayer) [reply Sources/Spezi/Module/Capabilities/Communication/CollectPropertyWrapper.swift:19]: You are right, good idea! With the new SwiftSyntax usage in SwiftLint that could be entirely portable and maybe even reliable compared to some older rules.

File: Sources/Spezi/Configuration/Component.swift (hunk 4 above)
Comment (PSchmiedmayer) [reply Sources/Spezi/Configuration/Component.swift:58]: The problem that I would see is if the `Standard` is not yet fully configured at that stage, we would need to first enforce that all `Dependencies` of the Standard are configured? That would require some additional logic in the `Spezi` types. If we somehow enforce that all `@Dependencies` of the `Standard` are configured we will then have to ensure that they are called before and then we are going to have a race condition with the `prepare()` function or no real definition when it is called.

Given these thoughts, I would slightly favor removing the function and adding it at a point where we have a clear use case that goes beyond what we can do in the initializer?

File: Sources/Spezi/Utilities/AnyOptional.swift (hunk 9 above)
Comment (PSchmiedmayer) [reply Sources/Spezi/Utilities/AnyOptional.swift:16]: Sure, that works for me, could be useful or some other use cases as well 👍

File: Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift (hunk 12 above)
Comment (PSchmiedmayer) [reply Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift:17]: Sounds good 🚀

File: Sources/Spezi/Configuration/Component.swift (hunk 4 above)
Comment (Supereg) [reply Sources/Spezi/Configuration/Component.swift:58]: Agree, removed 👍 Thanks for the detailed explanation.

File: Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift (hunk 13 above)
Comment (Supereg) [reply Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift:38]: I tried to somehow add a runtime check but failed with all my attempts. It's really sad that Swift won't allow you to check for `is Sendable`.

File: Sources/Spezi/Configuration/Component.swift
[the remark this answers is not in the document]
Comment (Supereg) [reply Sources/Spezi/Configuration/Component.swift:1]: As we removed the prepare method. Do you think this article is still a requirement for the PR?

File: Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift (hunk 12 above)
Comment (Supereg) [reply Sources/Spezi/SharedRepository/Builtin/ValueRepository.swift:17]: I added a Shared Repository article to DoCC and also made sure that all types related to Shared Repository are grouped accordingly.

File: Sources/Spezi/Configuration/Component.swift
[the remark this answers is not in the document]
Comment (PSchmiedmayer) [reply Sources/Spezi/Configuration/Component.swift:1]: Think it would still be great if we can at least add some additional documentation to the `Component` documentation what functionality should happen in the initializer, what in the `configure()` method, and what can then be handled by the `LifecycleHandler` protocol. Doesn't need to be a full article if you add this in the inline documentation here, up to you. Some guidance is surely greatly appreciated, especially now that we have to also consider collect and provide functionalities. I think DocC also supports adding additional context to a type using an article.

**Comment by PSchmiedmayer** [comment]
Thank you! I will have some time in the late afternoon to re-review the PR. If it is fine with you I can fix smaller issues that I might find myself and then merge the PR after that? Maybe I don't even have some feedback and could also directly merge it 👍

File: Sources/Spezi/Configuration/Component.swift
[the remark this answers is not in the document]
Comment (Supereg) [reply Sources/Spezi/Configuration/Component.swift:1]: I extracted current documentation of a `Component` into a separate Article Extension. I added a section about component communication. 
Further, I also slightly updated the organization of DocC topics.
We currently have a bit of duplication of the `Component` documentation and the `ModuleCapabilities.md` article. I think, this should receive some optimization. Maybe you can address this in #73 already.

[the remark this answers is not in the document]
Comment (PSchmiedmayer) [reply Sources/Spezi/Configuration/Component.swift:1]: Thank you, I will address this in #73 🚀

**Review by PSchmiedmayer — APPROVED** [review]
Amazing job with the PR @Supereg, very nice additions and a superb code quality!
